katdal/concatdata.py

```diff
@@ -233,11 +232,13 @@ def __init__(self, data):
 
     def __getitem__(self, key):
         """Extract timestamp, value and status of each sensor data point."""
-        return np.concatenate([sd[key] for sd in self.data])
+        return np.concatenate([sd[key] for sd in self._data])
 
-    def __len__(self):
-        """Number of sensor data points."""
-        return np.sum([len(sd) for sd in self.data])
+    def __bool__(self):
+        """True if sensor has at least one data point."""
+        return any([bool(sd) for sd in self._data])
```

Comment:
```sd``` is actually a SensorData object with a recarray-like interface. It used to also take an ```h5py.Dataset``` directly but I don't see that anywhere in the code anymore. So the chances are low that ```sd``` will be an ndarray.